### ISSUE TITLE
Sync on push for large project takes a few minutes on Win10

### ISSUE THREAD

**Issue by scottkurz** [openshift/odo#5100]
/kind bug

## What versions of software are you using?

**Operating System:**  Win 10, (also tried on Linux to demonstrate this was NOT a problem on Linux)

**Output of `odo version`:**

odo v2.3.1 (2ae31fb66)
Kubernetes: v1.20.0+bafe72f

## How did you run odo exactly?
(assumes project set)
1.  git clone git@github.com:scottkurz/openshift-workshop-was.git
2. cd openshift-workshop-was/labs/Openshift/RuntimeModernization/app
3. odo create mycomp
4. odo push --show-log -v 1

## Actual behavior

On Win10, I get something like 2 min here or more:
>  V  Syncing files to the component [2m]

## Expected behavior

On Linux, I get 8s for the same app
>   ✓  Syncing files to the component [8s]

## Any logs, error output, etc?

Attaching logs.. I did a separate run with `-v 9` output.  The above measurements (8s vs. 2m) were with `-v 1`, but I'm sure `-v 9` will slow things down even more.

[logs.zip](https://github.com/openshift/odo/files/7253306/logs.zip)

**Comment by awisniew90** [comment]
On Mac I'm seeing about ~1min for this app:

```
 ✓  Syncing files to the component [1m]
 ```

**Comment by kadel** [comment]
I've tested your scenario against remote OpenShift using Developer Sandbox (https://developers.redhat.com/developer-sandbox):
- Windows 10: `Syncing files to the component [25s]`
- Linux Subsystem on Windows 10 `Syncing files to the component [26s]`
- Directly on Linux `Syncing files to the component [24s]`

The application is 68Mb and times that I can see are matching my upload internet speed. 

I haven't built the application locally, so there was no `target` directory.


In attached logs I can see that content of the `target` directory was synced to the component.
It looks like longer sync times on windows are due to the #4389

**Comment by scottkurz** [comment]
OK... I removed `target` AND excluded the path from Windows antivirus and now am down to 33s.   @awisniew90 did you sync from a clean local HFS ?

**Comment by awisniew90** [comment]
Yep - mine was without a target dir. Consistently ~1m, but could be my local internet speeds...

**Comment by kadel** [comment]
I was not able to reproduce this with odo v3

testing on springboot petclinic sample

windows - sync 3s

mac - sync 3s

/close
